workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FAB.tsx

```diff
@@ -16,49 +16,104 @@
 
 import * as React from 'react';
 import { useNavigate } from 'react-router-dom';
+import { makeStyles } from '@mui/styles';
 import Fab from '@mui/material/Fab';
 import Tooltip from '@mui/material/Tooltip';
 import Typography from '@mui/material/Typography';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
 import { FabIcon } from './FabIcon';
 import { FloatingActionButton, Slot } from '../types';
 
+const useStyles = makeStyles(() => ({
+  openInNew: { paddingBottom: '5px', paddingTop: '3px' },
+}));
+
 export const FAB = ({
   actionButton,
   size,
+  className,
 }: {
   actionButton: FloatingActionButton;
   size?: 'small' | 'medium' | 'large';
+  className?: string;
 }) => {
   const navigate = useNavigate();
+  const styles = useStyles();
   const isExternalUri = (uri: string) => /^([a-z+.-]+):/.test(uri);
-  const external = isExternalUri(actionButton.to!);
-  const newWindow = external && !!/^https?:/.exec(actionButton.to!);
+  const isExternal = isExternalUri(actionButton.to!);
+  const newWindow = isExternal && !!/^https?:/.exec(actionButton.to!);
   const navigateTo = () =>
-    actionButton.to && !external ? navigate(actionButton.to) : '';
+    actionButton.to && !isExternal ? navigate(actionButton.to) : '';
+  const labelText =
+    actionButton.label.length > 20
+      ? `${actionButton.label.slice(0, actionButton.label.length)}...`
+      : actionButton.label;
+  const getColor = () => {
+    if (actionButton.color) {
+      return actionButton.color;
+    }
+    if (!className) {
+      return 'info';
+    }
+    return undefined;
+  };
```

Comment:
Just asking myself: Why are you mixing className and color here?

Couldn't we use info always as fallback?

Reply:
@christoph-jerolimov  As per the UX the sub-menu actions should have the secondary color (ref: --pf-t--global--background--color--secondary--default , https://www.patternfly.org/design-foundations/colors/#background-colors) by default if the color is not specified. And for the main FAB button the default color should be blue or 'info'


cc @ShiranHi